chips/cc26x2/src/crt1.rs

```diff
@@ -14,6 +17,56 @@ extern "C" {
     fn _estack();
 }
 
+use events;
+macro_rules! generic_isr {
+    ($label:tt, $priority:expr) => {
+        #[cfg(target_os = "none")]
+        #[naked]
+        unsafe extern "C" fn $label() {
+            stash_process_state();
+            asm!("
+                // Set event flag
+                orr $0, $2
+                isb
+                "
+                : "={r0}"(events::EVENTS)
+                : "{r0}"(events::EVENTS), "{r1}"(0b1<<($priority as u8))
+                : : "volatile" "volatile"
+            );
```

Comment:
We might want to consider adding `cpsid i` and `cpsie i` around this `asm!`. Because NVIC interrupts are nested, there could be another high priority interrupt that could potentially overwrite `events::EVENTS`, resulting in lost interrupts.

Following is a snippet that I am currently prototyping to learn more about your design.

```
          1:
            /* Due to the nested nature of NVIC interrupts, we need to disable   */
            /* interrupts before making changes to INTERRUPT_EVENTS. Once we are */
            /* done setting the appropriate bit, we enable the interrupts again  */
            cpsid i

            ldr r0, =INTERRUPT_EVENTS
            ldr r1, [r0, #0]
            orr r1, $0
            str r1, [r0, #0]

            cpsie i

            bx lr"
            :
            : "{r2}"(0b1 << ($interrupt as u8))
            : : "volatile");
        }
```

Reply:
Although currently all NVICs are the same priority level and so they will not interrupt each other, they could potentially so I did protect this assembly block with cpsid's.